rules/autoconsent/ubuntu.com.json

```diff
@@ -0,0 +1,107 @@
+{
+    "name": "ubuntu.com",
+    "prehideSelectors": [
+        "dialog.cookie-policy"
+    ],
+    "detectCmp": [
+        {
+            "exists": [
+                [
+                    "aria/Your tracker settings[role=\"dialog\"]",
+                    "aria/[role=\"banner\"]"
+                ],
+                [
+                    "dialog.cookie-policy header"
+                ],
+                [
+                    "xpath///*[@id=\"modal\"]/div/header"
+                ],
+                [
+                    "pierce/dialog header"
+                ]
+            ]
```

Comment:
I find this structure a bit confusing. If I understand this right, this can be an array of arrays of strings, where the top-level elements are combined with an AND (all groups must match), but the lower level is essentially OR (we pick one selector from each group). It's quite unintuitive if you're not familiar with the Recorder API.
I wonder if we can come up with a more readable/explicit format.

Reply:
Is the other way around:
- The top level array are equivalent selectors. We iterate through them until one matches something, then return that.
- Within the nested array, we can chain selectors to make more specific rules. For example `["aria/Your tracker settings[role=\"dialog\"]", "aria/[role=\"banner\"]"]` means, looks for the "Your tracker settings" dialog, then under that element, find the element with `aria-role=banner`.

I agree this can be confusing at first, but there are also good reasons to not change it, namely:
- that this structure is directly compatible with multiple tools (Chrome recorder and puppeteer).
- it works nicely with our existing rule selectors (runtime detection between string and array is easy), so we can easily mix and match different selector formats.

I'm not sure creating our own custom format helps that much, as it is then just another new format that diverges from anything a development might have seen before.